lib/sequenceserver/job.rb

```diff
@@ -26,7 +25,9 @@ def create(params)
       #
       # TODO: What if the given job id does not exist?
       def fetch(id)
-        YAML.load_file(File.join(DOTDIR, id, 'job.yaml'))
+        job_file = File.join(DOTDIR, id, 'job.yaml')
+        fail BLAST::ArgumentError, "KeyError: Requested job was not found" unless File.exist?(job_file)
```

Comment:
`BLAST::ArgumentError` is an incorrect error class for this scenario. You could raise `KeyError` here or define a new error class - I think former is okay. The corresponding handler in routes.rb should return 404.